programs/vote/src/vote_state/handler.rs

```diff
@@ -634,6 +708,9 @@ impl VoteStateHandler {
             VoteStateTargetVersion::V3 => {
                 VoteStateV3::default().set_vote_account_state(vote_account)
             }
+            VoteStateTargetVersion::V4 => {
+                VoteStateV4::default().set_vote_account_state(vote_account)
+            }
```

Comment:
This should just zero the vote account data after we do the resize / rent check for prev vote state versions